### PR TITLE
Sidm 3585

### PR DESCRIPTION
**Pull request by radoslaw-orlowski-amido**
### Change description ###

Workaround for SLF4J logging not working for Health Checker.
As part of the investigation I have upgraded Gradle to the most recent version. It's not required for the changes but I decided to leave it as it is (upgraded) anyway.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,14 +2,16 @@ plugins {
     id 'java-library'
     id 'org.springframework.boot' version '2.0.6.RELEASE'
     id 'maven-publish'
-    id 'com.github.johnrengelman.shadow' version '2.0.3'
+    id 'com.github.johnrengelman.shadow' version '5.2.0'
     id 'com.jfrog.bintray' version '1.8.1'
+    id "io.freefair.lombok" version "5.0.0-rc4"
```
Comment (radoslaw-orlowski-amido) [inline build.gradle]: Explicit Lombok has been replaced by a dedicated plugin.

File: build.gradle [hunk 2]
```diff
@@ -2,14 +2,16 @@ plugins {
...
 }
 
 
 group = 'uk.gov.hmcts.reform'
 version = '2.0.2'
 sourceCompatibility = 1.8
+generateLombokConfig.enabled = false
```
Comment (radoslaw-orlowski-amido) [inline build.gradle]: Otherwise, the plugin will overwrite the file on each build.

File: build.gradle [hunk 3]
```diff
@@ -70,6 +72,12 @@ publishing {
     }
 }
 
+jar {
```
Comment (dfourn) [inline build.gradle]: Does this work with jar? 
Is the implementation-version present in the manifest of the built jar file?

If it doesn't work, then bootJar should be used instead.

Comment (radoslaw-orlowski-amido) [reply build.gradle]: This one doesn't. There is another entry in the `bootJar` section that works. I've moved everything manifest-related to `bootJar` for clarity.

File: src/main/java/uk/gov/hmcts/reform/idam/health/util/LogHelper.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    }
+
+    /**
+     * Returns a new logger associated with a given class. Used by lombok's {@link lombok.CustomLog CustomLog}.
+     *
+     * @param clazz the class
+     * @return the newly created logger
+     */
+    @SuppressWarnings("unused")
```
Comment (dfourn) [inline src/main/java/uk/gov/hmcts/reform/idam/health/util/LogHelper.java]: This is probably not required since this method is used in a test.

**Pull request merged**